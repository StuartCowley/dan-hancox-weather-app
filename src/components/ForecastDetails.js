import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecast }) {
  const { date, temperature, wind, humidity } = forecast;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__temperature">
        <div>Min Temp: {temperature.min}&deg;C</div>
        <div>Max Temp: {temperature.max}&deg;C</div>
      </div>
      <div className="forecast-details__wind">
        <div>Wind Speed: {wind.speed}mph</div>
        <div>Wind Direction: {wind.direction}</div>
      </div>
      <div className="forecast-details__humidity">Humidity: {humidity}</div>
    </div>
  );
}

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number,
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }).isRequired,
  }).isRequired,
};
export default ForecastDetails;
