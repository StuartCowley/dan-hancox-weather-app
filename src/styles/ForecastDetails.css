.forecast-details {
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 10px;
  background-color: #f2f2f2;
  border-radius: 5px;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
  margin-top: 10px;
}

.forecast-details__date {
  font-size: 24px;
  font-weight: bold;
  color: #333;
  margin-bottom: 10px;
}

.forecast-details__temperature {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  font-size: 18px;
  color: #333;
  margin-bottom: 10px;
}

.forecast-details__temperature div {
  margin: 0 5px;
}

.forecast-details__wind,
.forecast-details__humidity {
  font-size: 14px;
  color: #333;
  margin-bottom: 10px;
}
