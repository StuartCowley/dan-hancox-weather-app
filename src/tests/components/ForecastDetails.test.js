import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    humidity: 60,
    temperature: {
      max: 22,
      min: 12,
    },
    wind: {
      speed: 80,
      direction: "e",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
