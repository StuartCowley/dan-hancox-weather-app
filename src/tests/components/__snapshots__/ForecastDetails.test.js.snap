// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu Jan 01 1970
    </div>
    <div
      class="forecast-details__temperature"
    >
      <div>
        Min Temp: 12°C
      </div>
      <div>
        Max Temp: 22°C
      </div>
    </div>
    <div
      class="forecast-details__wind"
    >
      <div>
        Wind Speed: 80mph
      </div>
      <div>
        Wind Direction: e
      </div>
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 60
    </div>
  </div>
</DocumentFragment>
`;
